### PR TITLE
[DOCS] Rename mount types for searchable snapshots

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Renames 'full copy searchable snapshot' to 'fully mounted index.'
* Renames 'shared cache searchable snapshot' to 'partially mounted index.'
* Removes some unneeded cache setup instructions for the frozen tier. We added a default cache size with #71844.

### Previews
https://elasticsearch_72699.docs-preview.app.elstc.co/diff

### CODE REVIEW

File: docs/reference/datatiers.asciidoc [hunk 1]
```diff
@@ -88,20 +88,15 @@ For resiliency, indices in the cold tier can rely on
...
-must sometimes fetch data from the snapshot repository.
+The frozen tier uses <<searchable-snapshots,partially mounted {search-snaps}>>
+to store and load data from a snapshot repository. When an index transitions into
+the <<ilm-index-lifecycle,frozen phase>>, {ilm-init} automatically converts it
+to a partially mounted {search-snap} on the frozen tier.
+
+Partially mounted {search-snaps} let you search over large data sets with fewer
+compute resources, reducing both your storage and operating costs. Because {es}
+must sometimes fetch data from the snapshot repository, searches on the frozen
+tier are typically slower than on the cold tier.
```
Comment (jrodewig) [inline docs/reference/datatiers.asciidoc:99]: I refactored this section to deduplicate content already covered in the mount options section of the searchable snapshots page.

File: docs/reference/searchable-snapshots/apis/mount-snapshot.asciidoc [hunk 2]
```diff
@@ -48,12 +48,18 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=master-timeout]
...
+--
+(Optional, string)
+<<searchable-snapshot-mount-storage-options,Mount option>> for the
+{search-snap} index. Possible values are:
+
+`full_copy` (Default):::
+<<fully-mounted,Fully mounts>> the index.
+
+`shared_cache`:::
+<<partially-mounted,Partially mounts>> the index.
```
Comment (jrodewig) [inline docs/reference/searchable-snapshots/apis/mount-snapshot.asciidoc:61]: Refactored to deduplicate content from the mount options page.

File: docs/reference/searchable-snapshots/index.asciidoc [hunk 3]
```diff
@@ -120,39 +120,38 @@ To search a snapshot, you must first mount it locally as an index. Usually
 are two options for mounting a snapshot, each with different performance
 characteristics and local storage footprints:
 
-[[full-copy]]
-Full copy::
+[[fully-mounted]]
```
Comment (jrodewig) [inline docs/reference/searchable-snapshots/index.asciidoc:123]: As a general note, I think this page could use a refactor to reorder the content and deduplicate. However, that work is less pressing and outside the scope of this PR. I opened https://github.com/elastic/elasticsearch/issues/72728 to track.

File: docs/reference/tab-widgets/data-tiers.asciidoc [hunk 4]
```diff
@@ -36,19 +36,8 @@ node.roles: [ data_cold ]
...
-
-[source,yaml]
-----
-node.roles: [ data_frozen ]
-xpack.searchable.snapshot.shared_cache.size: 50GB
-----
-
-If needed, you can assign a node to more than one tier.
+We recommend you use dedicated nodes in the frozen tier. If needed, you can
+assign other nodes to more than one tier.
```
Comment (jrodewig) [inline docs/reference/tab-widgets/data-tiers.asciidoc:40]: I simplified this based on the new default settings.

**Review by henningandersen — COMMENTED** [review]
Thanks for working on this. The naming decision we made was to call these indices "Partially mounted indices" and "Fully mounted indices". The naming change done here seems to use "Partially mounted searchable snapshots" and "Fully mounted searchable snapshots". Can you update the PR to use the "indices" terminology?

**Comment by jrodewig** [comment]
Thanks for the clarification @henningandersen. I've revised throughout to use "partially mounted index" in place of "partially mounted searchable snapshot." This is ready for another look at your convenience.

**Review by henningandersen — APPROVED** [review]
LGTM, thanks @jrodewig

**Comment by jrodewig** [comment]
Thanks @henningandersen!

**Pull request merged**